File: targets/python-flask/routes/render.py
"""Rendering routes."""
from flask import Blueprint, request, render_template_string

render_bp = Blueprint("render", __name__, url_prefix="/render")


@render_bp.route("/preview")
def preview():
    tpl = request.args.get("tpl", "")
    return render_template_string(tpl)


@render_bp.route("/hello")
def hello():
    name = request.args.get("name", "")
    return "<h1>Hello " + name + "</h1>"

File: targets/js-ts-express/routes/render.js
// Rendering routes.
const express = require('express');
const ejs = require('ejs');

const router = express.Router();

router.post('/preview', (req, res) => {
  const tpl = req.body.template;
  const html = ejs.render(tpl, { user: req.body.user });
  res.send(html);
});

router.get('/hello', (req, res) => {
  const name = req.query.name;
  res.send('<h1>Hello ' + name + '</h1>');
});

module.exports = router;
